File: backend/telemetry-bff/domain/devicesManager/entities/devices.go
package entities

type LocationType string

const (
	Point LocationType = "Point"
)

type Location struct {
	Type        LocationType `json:"type,omitempty"`
	Coordinates []float64    `json:"coordinates,omitempty"`
}
type DeviceData struct {
	SensorId   string   `json:"sensor_id,omitempty" bson:"sensor_id"`
	Location   Location `json:"location,omitempty" bson:"location"`
	TypeSensor string   `json:"type_sensor,omitempty" bson:"type_sensor"`
	Units      []string `json:"units,omitempty" bson:"units"`
}

File: frontend/telemetry-ui/src/pages/devices/entities/devices.ts
export interface Location {
  lat: number;
  long: number;
}

export interface DeviceLocation {
  type: string;
  coordinates: Array<number>;
}

export interface DeviceInfo {
  sensor_id: string;
  location: DeviceLocation;
  type_sensor: string;
  units: Array<string>;
}

export interface DeviceData {
  last_ping: Date;
  device: DeviceInfo;
}
